firedrake/assemble.py

```diff
@@ -20,7 +20,7 @@
 
 
 def assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
-             inverse=False, nest=None):
+             inverse=False, nest=None, matfree=False, extra_ctx={}):
```

Comment:
Agreed.  This also provides a possible hook for assembling
coordinate-format matrices (MATIS) that PETSc's BDDC preconditioners want
-- these are supposed to be quite powerful, though I haven't looked at them.

Do you want to get this, or shall I?

On Mon, Aug 15, 2016 at 9:22 AM, Lawrence Mitchell <notifications@github.com
> wrote:

> In firedrake/assemble.py
> <https://github.com/firedrakeproject/firedrake/pull/853#discussion_r74768528>
> :
>
> > @@ -20,7 +20,7 @@
> >
> >
> >  def assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
> > -             inverse=False, nest=None):
> > +             inverse=False, nest=None, matfree=False, extra_ctx={}):
>
> There are now two different arguments that control the type of the
> resulting matrix. We should merge these into:
>
> mat_type=["aij", "nest", "matfree"]
>
> Then there isn't the same thing of "does nest override matfree or not".
> The nest argument then disappears from the variational solvers and turns up
> in the solver_parameters instead.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/firedrakeproject/firedrake/pull/853/files/ff5ec4e0eabb94576ac69c9c82932eaa50b2d14e#r74768528>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADl6L9o2PbC2Z8HeFcTNwlIkzK8js0RDks5qgHYtgaJpZM4JiV-g>
> .
>